File: test-utilities/src/main/scala/com/taskadapter/connector/testlib/ResourceLoader.scala
package com.taskadapter.connector.testlib

object ResourceLoader {

  def getAbsolutePathForResource(name: String) = {
    val url = ResourceLoader.getClass.getClassLoader.getResource(name)
    try
      url.toURI.getPath
    catch {
      case e: Exception =>
        throw new RuntimeException(e)
    }
  }
}

File: test-utilities/src/main/java/com/taskadapter/connector/testlib/ResourceLoader.java
package com.taskadapter.connector.testlib;

public class ResourceLoader {
    public static String getAbsolutePathForResource(String name) {
        var url = ResourceLoader.class.getClassLoader().getResource(name);
        try {
            return url.toURI().getPath();
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}
